DeepSpeech.py

```diff
@@ -548,22 +577,83 @@ def train():
             init_op = tfv1.variables_initializer(missing_variables)
             session.run(init_op)
             loaded = True
-
-        tfv1.get_default_graph().finalize()
-
+            
         if not loaded and FLAGS.load in ['auto', 'last']:
+            tf.initialize_all_variables().run()
             loaded = try_loading(session, checkpoint_saver, 'checkpoint', 'most recent')
         if not loaded and FLAGS.load in ['auto', 'best']:
-            loaded = try_loading(session, best_dev_saver, 'best_dev_checkpoint', 'best validation')
-        if not loaded:
-            if FLAGS.load in ['auto', 'init']:
-                log_info('Initializing variables...')
-                session.run(initializer)
-            else:
-                log_error('Unable to load %s model from specified checkpoint dir'
-                          ' - consider using load option "auto" or "init".' % FLAGS.load)
-                sys.exit(1)
-
+            tf.initialize_all_variables().run()
+            loaded = try_loading(session, best_dev_saver, 'best_dev_checkpoint', 'best validation') 
+        if not loaded and FLAGS.load == "transfer":
+            if FLAGS.source_model_checkpoint_dir:
+                print('Initializing model from', FLAGS.source_model_checkpoint_dir)
+                ckpt = tfv1.train.load_checkpoint(FLAGS.source_model_checkpoint_dir)
+                variables = list(ckpt.get_variable_to_shape_map().keys())
+
+                # Load desired source variables
+                for v in tf.global_variables():
+                    # only load variables we do not drop from source model
+                    if not any(layer in v.op.name for layer in drop_source_layers):
+                        print('Loading from source model layer:', v.op.name)
+                        v.load(ckpt.get_tensor(v.op.name), session=session)
+                    else:
+                        print('Dropping from source model layer:', v.op.name)
+                        
+                # Initialize all variables needed for DS, but not loaded from ckpt
+                init_op = tfv1.variables_initializer(
+                    [v for v in tf.global_variables()
+                     if any(layer in v.op.name
+                            for layer in drop_source_layers)
+                     ])
+                session.run(init_op)
+            elif FLAGS.cudnn_source_model_checkpoint_dir:
```

Comment:
I'd love if we can just re-use `--use_cudnn_rnn` here, with `--source_model_checkpoint_dir`